### DISCUSSION TITLE
Proposal: Batch Automated 3D Printing with Auto-Ejection

### DISCUSSION THREAD

**Discussion by Neckr0ik** [OrcaSlicer/OrcaSlicer#7693] [Ideas]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Proposal: Batch Automated 3D Printing for OrcaSlicer</title>
</head>
<body style="font-family: Arial, sans-serif; line-height: 1.6; margin: 20px;">
    <h1>Proposal: Batch Automated 3D Printing with Auto-Ejection</h1>
    <p>
        Dear OrcaSlicer Community and Developers,
    </p>
    <p>
        I’d like to propose a new feature for OrcaSlicer that could significantly enhance its capabilities: 
        <strong>Batch Automated 3D Printing with Auto-Ejection Support</strong>. This feature aims to streamline 
        the printing of multiple parts by automating the process of printing, ejecting, and preparing for the next print.
    </p>
    <h2>Key Features</h2>
    <ul>
        <li>
            <strong>Batch Printing Workflow:</strong>
            <ul>
                <li>Allow users to specify the number of prints required.</li>
                <li>Automatically duplicate and arrange parts on the print bed.</li>
            </ul>
        </li>
        <li>
            <strong>User-Defined Input Parameters:</strong>
            <ul>
                <li><strong>Model Height Analysis:</strong> Use the sliced model height to determine feasibility for auto-ejection.</li>
                <li>
                    <strong>3D Printer Mode:</strong> Adapt movements based on printer type:
                    <ul>
                        <li><strong>Core XY:</strong> Execute precise lateral ejection movements.</li>
                        <li><strong>Core XZ:</strong> Consider vertical eject adjustments.</li>
                    </ul>
                </li>
            </ul>
        </li>
        <li>
            <strong>Auto-Ejection G-code Integration:</strong>
            <ul>
                <li>Enable users to input custom G-code for bed clearing after each print.</li>
                <li>Provide pre-configured templates for popular printers (e.g., Ender, Bambu Lab X1C).</li>
            </ul>
        </li>
        <li>
            <strong>Visual Preview of Auto-Ejection:</strong>
            <ul>
                <li>Include a simulation showing the ejection sequence, so users can validate their setup before initiating a print job.</li>
            </ul>
        </li>
        <li>
            <strong>Smart Plate Management:</strong>
            <ul>
                <li>Automatically cool or heat the bed as required for safe and effective part removal.</li>
                <li>Include checks for bed adhesion and material compatibility.</li>
            </ul>
        </li>
        <li>
            <strong>Failsafes and Triggers:</strong>
            <ul>
                <li>Analyze print settings to ensure successful ejection (e.g., minimum part height, adhesion strength).</li>
                <li>Abort ejection if risks are detected (e.g., part height too small, risk of damage).</li>
            </ul>
        </li>
    </ul>
    <h2>Benefits</h2>
    <ul>
        <li><strong>Efficiency:</strong> Automates tedious manual tasks, allowing continuous printing without user intervention.</li>
        <li><strong>Scalability:</strong> Ideal for producing multiple identical parts or small batch production.</li>
        <li><strong>Compatibility:</strong> Expands OrcaSlicer’s utility for advanced users and automation enthusiasts.</li>
        <li><strong>Preview Confidence:</strong> Users can test and adjust ejection strategies with visual feedback.</li>
    </ul>
    <h2>Technical Challenges</h2>
    <ul>
        <li><strong>Printer Compatibility:</strong> Not all printers support automated part ejection; hardware prerequisites must be defined.</li>
        <li><strong>G-code Complexity:</strong> Developing robust and flexible G-code templates for diverse printer models.</li>
        <li><strong>Safety Considerations:</strong> Ensuring no damage to parts, printers, or eject mechanisms.</li>
    </ul>
    <h2>Call to Action</h2>
    <p>
        I believe this feature would position OrcaSlicer as the go-to slicer for automated workflows, competing strongly with both proprietary and open-source solutions. 
        If the community supports this idea, we could collaboratively refine the concept and create a roadmap for development.
    </p>
    <p>
        I look forward to hearing your thoughts and suggestions!
    </p>
    <p>
        Best regards,<br>
        <strong>Neckr0ik</strong>
    </p>
</body>
</html>

**Comment by EnygmaINC** [comment]
Curious - how would the ejection occur, exactly?

**Reply by Neckr0ik** [reply]
My idea on a bedslinger is using the X axis to push the object forward first, then using the toolhead itself just to make a second pass to certify the bed plate is clear, here's something I use already between my prints

; Auto-Extraction GCODE
M140 S45 ; Set bed temperature to 45°C
M190 S45 ; Wait for bed temperature to reach 45°C

; --- Variables ---
{bed_x} = 256 ; Total bed width (X)
{bed_y} = 262.5 ; Total bed depth (Y)
{safe_z} = 40 ; Safe Z height
{center_x} = {{bed_x} / 2} ; Center of the bed on X axis

; Positioning for extraction
G90 ; Set absolute positioning

; --- Start pushing sequence at the left edge ---
G1 Z{safe_z} F1200 ; Move Z to safe height
G1 X40 Y{bed_y} F3000 ; Move to X: 40 (left edge), Y: max (back)
G1 Z1.5 F600 ; Lower Z to 1.5mm
G1 Y0 F1000 ; Push forward (Y min)
G1 Y{bed_y} F3000 ; Return to Y max (back)

; --- Pushing sequence at the center of the bed ---
G1 Z{safe_z} F1200 ; Move Z to safe height
G1 X{center_x} F3000 ; Move to the center of the bed (X axis)
G1 Z1.5 F600 ; Lower Z to 1.5mm
G1 Y0 F1000 ; Push forward (Y min)
G1 Y{bed_y} F3000 ; Return to Y max (back)

; --- Finish extraction ---
G1 Z{safe_z} F1200 ; Move Z to safe height
G1 Y{bed_y} F3000 ; Return to Y max (safe position)
M400 ; Wait for all movements to finish

**Comment by DARKFiB3R** [comment]
Make this easier...

https://www.youtube.com/watch?v=Vxj1ii6dPYo

**Comment by hexsmith** [comment]
It's been done:

https://forum.bambulab.com/t/multi-plate-automation-for-a1-mini/79481
